test-e2e/server/api.spec.ts

```diff
@@ -1,140 +1,140 @@
+  describe('test valid request with jpeg', () => {
+    let jpegFormData: FormData
+
+    beforeEach(async () => {
+      const fileBuffer = await readFile(
+        path.resolve(pathUtil.testAsset, 'wtm_256x256.jpg'),
+      )
+      jpegFormData = new FormData()
+      jpegFormData.append('file', fileBuffer, {
+        contentType: 'image/jpg',
+        filename: 'wtm_256x256.jpg',
+      })
     })
-    const json = await fetched.json()
-    const jobId = parseInt(json['jobId'])
 
-    expect(fetched.ok).toEqual(true)
-    expect(jobId).toBeGreaterThan(0)
-    expect(jobId).toBeLessThan(1000)
-  })
+    it('POST /jobs convert format', async () => {
+      const config: IServerImageHandlerConfig = {
+        inputFormat: EnumFileFormat.jpg,
+        outputFormat: EnumFileFormat.png,
+      }
+      jpegFormData.append('config', JSON.stringify(config))
 
-  it('POST /jobs resize', async () => {
-    const fileBuffer = await readFile(
-      path.resolve(pathStore.testAsset, 'wtm_256x256.jpg'),
-    )
-    const config: IServerImageHandlerConfig = {
-      inputFormat: EnumFileFormat.jpg,
-      width: 128,
-      height: 128,
-    }
-    const formData = new FormData()
-    formData.append('file', fileBuffer, {
-      contentType: 'image/jpg',
-      filename: 'wtm_256x256.jpg',
-    })
-    formData.append('config', JSON.stringify(config))
-    const fetched = await fetch(`${apiHost}/jobs`, {
-      method: 'post',
-      body: formData,
+      const fetched = await fetch(`${apiHost}/jobs`, {
+        method: 'POST',
+        body: jpegFormData,
+        headers: authHeaders,
+      })
+      const json = await fetched.json()
+      const jobId = parseInt(json['jobId'])
+
+      expect(fetched.status).toEqual(201)
+      expect(jobId).toBeGreaterThan(0)
+      expect(jobId).toBeLessThan(1000)
     })
-    const json = await fetched.json()
-    const jobId = parseInt(json['jobId'])
 
-    expect(fetched.ok).toEqual(true)
-    expect(jobId).toBeGreaterThan(0)
-    expect(jobId).toBeLessThan(1000)
-  })
+    it('POST /jobs resize', async () => {
+      const config: IServerImageHandlerConfig = {
+        inputFormat: EnumFileFormat.jpg,
+        width: 128,
+        height: 128,
+      }
+      jpegFormData.append('config', JSON.stringify(config))
 
-  it('POST /jobs optimize', async () => {
-    const fileBuffer = await readFile(
-      path.resolve(pathStore.testAsset, 'wtm_256x256.jpg'),
-    )
-    const config: IServerImageHandlerConfig = {
-      inputFormat: EnumFileFormat.jpg,
-      optimizeAlgo: EnumOptimizationAlgorithm.mozjpeg,
-      quality: 90,
-    }
-    const formData = new FormData()
-    formData.append('file', fileBuffer, {
-      contentType: 'image/jpg',
-      filename: 'wtm_256x256.jpg',
-    })
-    formData.append('config', JSON.stringify(config))
-    const fetched = await fetch(`${apiHost}/jobs`, {
-      method: 'post',
-      body: formData,
+      const fetched = await fetch(`${apiHost}/jobs`, {
```

Comment:
I didn't switch to `supertest`, though it would be nice to not have to work with multiple testing strategies as I switch between projects.